### PR TITLE
User docs for AppProfiles (a.k.a. gen2 profiles and apps)

### PR DESCRIPTION
**Pull request by bcle**

### CODE REVIEW

**Review by Rohitrajak1807 — CHANGES_REQUESTED** [review]
Just one tiny change

File: docs/appprofiles.md [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+updates the `spec.generators` section of each affected app to target the cluster. The set of apps targeting a cluster is the union
+of all the valid appNames from every app profile listed in the annotation.
+
+By default, every ArgoCD Application Resource generated by the app's ApplicationSet is named with the pattern `<clustername>-app-<appname>`.
+In the running example, these two ArgoCD application resources will be generated:
+- clust1-app-wordpress
+- clust1-app-myconfigmap
+
+To detach all profiles from a cluster, simply remove the annotation:
+```
```
Comment (cruizen) [inline docs/appprofiles.md:172]:
```suggestion
```shell
```

File: docs/appprofiles.md [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+To indicate whether some app names are currently invalid, the AppProfile controller
+will update the resource's `status` section as follows:
+- If all specified app names refer to valid Arlon apps, `status.health`
+  is set to `healthy`.
+- If one or more specified app names refer to non-existent Arlon apps,
+  then `status.health` is set to `degraded`, and `status.invalidAppNames`
+  lists the invalid names.
+
+Here is an example of an AppProfile manifest that includes 3 apps, one of which does not exist:
+```
```
Comment (cruizen) [inline docs/appprofiles.md:120]:
```suggestion
```yaml
```

File: docs/appprofiles.md [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+While it is possible for you create and edit an ApplicationSet resource manifest
+satisfying the requirements to be an Arlon App from scratch, Arlon makes
+this easier with the `arlon app create --output-yaml` command, which outputs an
+initial compliant manifest that you can
+save to a file and edit to your liking before applying to
+the management cluster to actually create the app.
+(Without the ``--output-yaml` flag, the command will apply the resource for you).
+
+Here's an example of an initial Arlon Application manifest:
+```
```
Comment (cruizen) [inline docs/appprofiles.md:43]:
```suggestion
```yaml
```

File: docs/appprofiles.md [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+to ensure that any actual ArgoCD Application resources deployed from the ApplicationSet
+are uniquely named, by prefixing the cluster name.
+- `spec.template.spec.destination.server` to `{{cluster_server}}` to target the correct workload cluster
+
+Arlon apps can be listed in two ways. The first is to use the `arlon app list` command. One advantage
+is that it's simple to use and also displays additional information about the app, such as which AppProfiles
+are currently associated with the app.
+
+Example:
+```
```
Comment (cruizen) [inline docs/appprofiles.md:87]:
```suggestion
```shell
```

**Review by cruizen — APPROVED** [review]
Approved. Made some minor formatting suggestions.

**Pull request merged**